apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -87,76 +101,148 @@ export function positionBlocksOnWorkspace(workspace, blockOrderMap) {
     false
   );
 
-  orderedBlocksSetupFirst.forEach(block => {
-    positionBlockWithCursor(block, cursor);
+  adjustBlockPositions(orderedBlocksSetupFirst, workspace);
+}
+
+/**
+ * Positions blocks with any mix of user-defined positions and default positions (including all of one or the other)
+ * Such that none of the blocks overlap with each other
+ * @param {Array<Blockly.Block>} blocks - The blocks to position
+ * @param {Blockly.Workspace} workspace - The current Blockly workspace
+ */
+function adjustBlockPositions(blocks, workspace) {
+  let existingColliders = [];
```

Comment:
nit: it took me some time to realize `existingColliders` always was ordered top to bottom. Can we either add a comment explaining that here or maybe rename to `orderedColliders`?